tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/EagerOperationBuilder.java

```diff
@@ -60,23 +58,28 @@
     this.session = session;
     this.type = type;
     this.name = name;
-    this.nativeRef = new NativeReference(session, this, allocate(session.nativeHandle(), type));
+    this.opHandle = allocate(session.nativeHandle(), type);
+    session.attach(opHandle);
+    opHandle.releaseReference();
   }
 
   @Override
   public EagerOperation build() {
-    TFE_TensorHandle[] tensorHandles = execute(nativeRef.opHandle);
+    TFE_TensorHandle[] tensorHandles = execute(opHandle);
     EagerOperation operation =
-        new EagerOperation(session, nativeRef.opHandle, tensorHandles, type, name);
+        new EagerOperation(session, opHandle, tensorHandles, type, name);
     // Release our reference to the native op handle now that we transferred its
     // ownership to the EagerOperation
-    nativeRef.clear();
+    session.detach(opHandle);
```

Comment:
In `EagerOperation` constructor, we attach the `opHandle` to the session, which was already attached to it in the constructor of this class, before we detach it here. So if I understand correctly, attaching more than once the same resource to an `EagerSession` (therefore to a `PointerScope`) requires to detach it the same amount of time?

Reply:
Yes, that's how reference counting works.